redisvl/index/index.py

```diff
@@ -368,26 +376,27 @@ def connect(self, redis_url: Optional[str] = None, **kwargs):
 
         Args:
             redis_url (Optional[str], optional): The URL of the Redis server to
-                connect to. If not provided, the method defaults to using the
-                `REDIS_URL` environment variable.
+                connect to.
 
         Raises:
             redis.exceptions.ConnectionError: If the connection to the Redis
                 server fails.
             ValueError: If the Redis URL is not provided nor accessible
                 through the `REDIS_URL` environment variable.
+            ModuleNotFoundError: If required Redis modules are not installed.
 
         .. code-block:: python
 
             index.connect(redis_url="redis://localhost:6379")
 
         """
-        client = RedisConnectionFactory.connect(
-            redis_url=redis_url, use_async=False, **kwargs
+        # TODO: Intentionally not including required modules to match existing
+        # behavior, but we need to review.
+        self.__redis_client = RedisConnectionFactory.get_redis_connection(
```

Comment:
I believe the existing behavior still validated modules -- we used the `@setup_redis` decorator on `set_client`
https://github.com/redis/redis-vl-python/pull/280/files#diff-ebc75d7c32f45e2d6c0c7ea45a8ab92fddb96197967b64ab1572dec4efbcad4aL388